clients/unshaded/src/main/java/tachyon/client/file/FileSystem.java

```diff
@@ -0,0 +1,375 @@
+   * options.
+   *
+   * @param src a Tachyon path to mount the data to
+   * @param dst a UFS path to mount the data from
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws TachyonException if a Tachyon exception occurs
+   */
+  void mount(TachyonURI src, TachyonURI dst) throws IOException, TachyonException;
+
+  /**
+   * Mounts a UFS subtree to the given Tachyon path. The Tachyon path is expected not to exist as
+   * the method creates it. This method does not transfer any data or metadata from the UFS. It
+   * simply establishes the connection between the given Tachyon path and UFS path.
+   *
+   * @param src a Tachyon path to mount the data to
+   * @param dst a UFS path to mount the data from
+   * @param options options to associate with this operation
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws TachyonException if a Tachyon exception occurs
+   */
+  void mount(TachyonURI src, TachyonURI dst, MountOptions options)
+      throws IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #openFile(TachyonURI, OpenFileOptions)} with default options.
+   *
+   * @param path the file to read from
+   * @return a {@link FileInStream} for the given path
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given file does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  FileInStream openFile(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Opens a file for reading.
+   *
+   * @param path the file to read from
+   * @param options options to associate with this operation
+   * @return a {@link FileInStream} for the given path
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given file does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  FileInStream openFile(TachyonURI path, OpenFileOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #rename(TachyonURI, TachyonURI, RenameOptions)} with default
+   * options.
+   *
+   * @param src the path of the source, this must already exist
+   * @param dst the path of the destination, this path should not exist
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given file does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  void rename(TachyonURI src, TachyonURI dst)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Renames an existing Tachyon path to another Tachyon path in Tachyon. This operation will be
+   * reflected in the underlying storage if the path is persisted.
```

Comment:
- do we require the rename operation happens within the same mount-point?
- "reflected in the underlying storage" --- not quite explicit what this means.